src/config.c

```diff
@@ -230,3 +303,115 @@ bool Config_Read(void)
+    WRITE_BOOL(enable_tr3_sidesteps);
+    WRITE_BOOL(enable_braid);
+    WRITE_BOOL(enable_compass_stats);
+    WRITE_BOOL(enable_total_stats);
+    WRITE_BOOL(enable_timer_in_inventory);
+    WRITE_BOOL(enable_smooth_bars);
+    WRITE_BOOL(enable_fade_effects);
+    WRITE_BOOL(fix_tihocan_secret_sound);
+    WRITE_BOOL(fix_pyramid_secret_trigger);
+    WRITE_BOOL(fix_secrets_killing_music);
+    WRITE_BOOL(fix_descending_glitch);
+    WRITE_BOOL(fix_wall_jump_glitch);
+    WRITE_BOOL(fix_bridge_collision);
+    WRITE_BOOL(fix_qwop_glitch);
+    WRITE_BOOL(fix_alligator_ai);
+    WRITE_BOOL(change_pierre_spawn);
+    WRITE_INTEGER(fov_value);
+    WRITE_INTEGER(resolution_width);
+    WRITE_INTEGER(resolution_height);
+    WRITE_BOOL(fov_vertical);
+    WRITE_BOOL(enable_demo);
+    WRITE_BOOL(enable_fmv);
+    WRITE_BOOL(enable_cine);
+    WRITE_BOOL(enable_music_in_menu);
+    WRITE_BOOL(enable_music_in_inventory);
+    WRITE_BOOL(enable_xbox_one_controller);
+    WRITE_BOOL(enable_round_shadow);
+    WRITE_BOOL(enable_3d_pickups);
+    WRITE_FLOAT(rendering.anisotropy_filter);
+    WRITE_BOOL(walk_to_items);
+    WRITE_BOOL(disable_trex_collision);
+    WRITE_INTEGER(start_lara_hitpoints);
+    WRITE_ENUM(healthbar_showing_mode, m_BarShowingModes);
+    WRITE_ENUM(airbar_showing_mode, m_BarShowingModes);
+    WRITE_ENUM(healthbar_location, m_BarLocations);
+    WRITE_ENUM(airbar_location, m_BarLocations);
+    WRITE_ENUM(enemy_healthbar_location, m_BarLocations);
+    WRITE_ENUM(healthbar_color, m_BarColors);
+    WRITE_ENUM(airbar_color, m_BarColors);
+    WRITE_ENUM(enemy_healthbar_color, m_BarColors);
+    WRITE_ENUM(screenshot_format, m_ScreenshotFormats);
+    WRITE_ENUM(ui.menu_style, m_UIStyles);
+    WRITE_INTEGER(maximum_save_slots);
+    WRITE_BOOL(revert_to_pistols);
+    WRITE_BOOL(enable_enhanced_saves);
+    WRITE_BOOL(enable_pitched_sounds);
+
+    // User settings
+    WRITE_BOOL(rendering.enable_bilinear_filter);
+    WRITE_BOOL(rendering.enable_perspective_filter);
+    WRITE_BOOL(rendering.enable_vsync);
+
+    // Not held in g_Config
+    json_object_append_int(root_obj, "hi_res", Screen_GetPendingResIdx());
+
+    WRITE_INTEGER(music_volume);
+    WRITE_INTEGER(sound_volume);
+    WRITE_INTEGER(input.layout);
+    WRITE_FLOAT(brightness);
+    WRITE_FLOAT(ui.text_scale);
+    WRITE_FLOAT(ui.bar_scale);
+
+    struct json_array_s *layout_arr = json_array_new();
+    for (INPUT_ROLE role = 0; role < INPUT_ROLE_NUMBER_OF; role++) {
+        json_array_append_int(
+            layout_arr, Input_GetAssignedScancode(INPUT_LAYOUT_USER, role));
+    }
+    json_object_append_array(root_obj, "layout_sdl", layout_arr);
+
+    struct json_value_s *root = json_value_from_object(root_obj);
+    char *data = json_write_pretty(root, "  ", "\n", &size);
+    json_value_free(root);
+
+    File_Write(data, sizeof(char), size - 1, fp);
+    File_Close(fp);
+    Memory_FreePointer(&data);
+
+    return true;
+}
```

Comment:
Missing end of the line character

Reply:
Resolved in [4ee0f5a](https://github.com/rr-/Tomb1Main/pull/637/commits/4ee0f5a3487067af292d756b533979a5b9ffd017)